src/languages/php.js

```diff
@@ -431,11 +430,59 @@ export default function(hljs) {
   };
   PARAMS_MODE.contains.push(FUNCTION_INVOKE);
 
+  const ATTRIBUTES = {
+    begin: regex.concat(/^\s*#\[\s*\\?/, PASCAL_CASE_CLASS_NAME_RE),
+    beginScope: "meta",
+    end: /]/,
+    endScope: "meta",
+    keywords: {
+      literal: LITERALS,
+      keyword: [
+        'new',
+        'array',
+      ]
+    },
+    contains: [
+      {
+        begin: /\[/,
+        end: /]/,
+        keywords: {
+          literal: LITERALS,
+          keyword: [
+            'new',
+            'array',
+          ]
+        },
+        contains: [
+          'self',
+          NAMED_ARGUMENT,
+          LEFT_AND_RIGHT_SIDE_OF_DOUBLE_COLON,
+          hljs.C_BLOCK_COMMENT_MODE,
+          STRING,
+          NUMBER,
+          CONSTRUCTOR_CALL,
+        ]
+      },
+      NAMED_ARGUMENT,
+      LEFT_AND_RIGHT_SIDE_OF_DOUBLE_COLON,
+      hljs.C_BLOCK_COMMENT_MODE,
+      STRING,
+      NUMBER,
+      CONSTRUCTOR_CALL,
+      {
+        scope: 'meta',
+        match: regex.concat(/\\?/, PASCAL_CASE_CLASS_NAME_RE)
+      }
+    ]
+  };
+
   return {
     case_insensitive: false,
     keywords: KEYWORDS,
     contains: [
-      hljs.HASH_COMMENT_MODE,
+      hljs.inherit(hljs.HASH_COMMENT_MODE, {
+        begin: /^\s*#(?!\[)/
+      }),
```

Comment:
Same issue with whitespace, why was this change necessary?

Reply:
Oh you're adding a guard... but would it work just fine without the whitespace?  This is usually solved with match order not negative look-aheads... we should be matching attributes before comments so the comment rule can't mismatch them... is there a reason we can't do that?

Reply:
> Oh you're adding a guard... but would it work just fine without the whitespace? This is usually solved with match order not negative look-aheads... we should be matching `ATTRIBUTE` before `COMMENT` so the comment rule can't mismatch them... is there a reason we can't do that?


Moving `ATTRIBUTES` above the old comment code was enough. Fixed.